tests/test_downloader_handlers.py

```diff
@@ -786,7 +786,10 @@ def test_download_with_proxy(self):
         def _test(response):
             self.assertEqual(response.status, 200)
             self.assertEqual(response.url, request.url)
-            self.assertEqual(response.body, b'http://example.com')
+            self.assertTrue(
+                response.body == b'http://example.com'  # HTTP/1.x
+                or response.body == b'/'  # HTTP/2
+            )
```

Comment:
What about keeping the old check, but turning `b'http://example.com'` into a class-level variable to be overridden in HTTP/2 tests? That way the check is more strict, and will fail if the protocol-version-specific expectations are not met (e.g. we suddenly start returning `/` for HTTP/1.x).